.github/workflows/integration_tests_pr.yml

```diff
@@ -49,15 +46,56 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ inputs.sha }}
+          fetch-depth: 0
+          submodules: 'recursive'
 
       - run: make deps
 
-      - run: make PKG_NAME="${{ inputs.module }}" int-test
-        if: ${{ steps.disallowed-character-check.outputs.result == 'pass' }}
+      - name: Update system packages
+        run: sudo apt-get update -y
+
+      - name: Install system deps
+        run: sudo apt-get install -y build-essential
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Install Python deps
+        run: pip3 install requests wheel boto3
+
+      - name: Install go-junit-report
+        run: go install github.com/jstemmer/go-junit-report/v2@latest
+
+      - name: Run Integration Tests
+        run: |
+          timestamp=$(date +'%Y%m%d%H%M')
+          report_filename="${timestamp}_terraform_test_report.xml"
+          if ! make int-test | go-junit-report -set-exit-code > "$report_filename"; then
+            echo "EXIT_STATUS=1" >> $GITHUB_ENV
+          fi
+          cat "$report_filename"
         env:
           LINODE_TOKEN: ${{ secrets.DX_LINODE_TOKEN }}
-          RUN_LONG_TESTS: ${{ inputs.run_long_tests }}
-      
+
+      - name: Add additional information to XML report
+        run: |
+          filename=$(ls | grep -E '^[0-9]{12}_terraform_test_report\.xml$')
+          python tod_scripts/add_to_xml_test_report.py \
+          --branch_name "${GITHUB_REF#refs/*/}" \
+          --gha_run_id "$GITHUB_RUN_ID" \
+          --gha_run_number "$GITHUB_RUN_NUMBER" \
+          --xmlfile "${filename}"
+
+      - name: Upload test results to bucket
+        run: |
+          report_filename=$(ls | grep -E '^[0-9]{12}_terraform_test_report\.xml$')
+          python3 tod_scripts/test_report_upload_script.py "${report_filename}"
+        env:
+          LINODE_CLI_OBJ_ACCESS_KEY: ${{ secrets.LINODE_CLI_OBJ_ACCESS_KEY }}
+          LINODE_CLI_OBJ_SECRET_KEY: ${{ secrets.LINODE_CLI_OBJ_SECRET_KEY }}
+
```

Comment:
Maybe we can leave this workflow simpler by not uploading to TOD and just showing the result on GitHub?

Reply:
It already reports the report in GitHub in the `Run integration Tests` step
`          cat "$report_filename"`